### PR TITLE
[Merged by Bors] - Improve contributing instructions

### PR DESCRIPTION
**Pull request by alice-i-cecile**
The existing contributing instructions are incomplete, and spread across the README and CONTRIBUTING.md.

This PR attempts to improve them and document the subtle process steps involved.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,6 +1,28 @@
...
+    2. `cd bevy-website`
+3. Start the Zola server with `zola serve`.
+
+A local server should start and you should be able to access a local version of the website from there.
+
+## Testing your changes
+
+The code in the book is compiled and tested to make sure that the examples work for readers.
+
+To check your local changes, move into the `code-validation` folder of this project using `cd code-validation`, and then run `cargo test`.
```
Comment (Nilirad) [inline CONTRIBUTING.md:23]:
```suggestion
To check your local changes, move into the `code-validation` folder of this project using `cd code-validation`, and then run `cargo test`. If you add a new section of the book, you have to add it in `code-validation/src/lib.rs` to enable code validation for it.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,6 +1,28 @@
...
+
+1. [Download Zola](https://www.getzola.org/).
+2. Clone the Bevy Website git repo and enter that directory:
+    1. `git clone https://github.com/bevyengine/bevy-website.git`
+    2. `cd bevy-website`
+3. Start the Zola server with `zola serve`.
+
+A local server should start and you should be able to access a local version of the website from there.
+
+## Testing your changes
```
Comment (Nilirad) [inline CONTRIBUTING.md:19]: IMO this heading should express that this paragraph is only about code snippets in the book.

Comment (alice-i-cecile) [reply CONTRIBUTING.md:19]:
```suggestion
## Testing changes to book code snippets
```

**Review by Nilirad — APPROVED** [review]

**Review by Weibye — APPROVED** [review]
This is much cleaner! 👍

**Review by IceSentry — APPROVED** [review]

**Pull request closed without merging**